Signal/src/ViewControllers/MessageComposeTableViewController.m

```diff
@@ -667,46 +675,49 @@ - (Contact *)contactForIndexPath:(NSIndexPath *)indexPath {
+        if ([_blockedPhoneNumbers containsObject:phoneNumber.toE164]) {
+            return YES;
+        }
+    }
+
+    return NO;
 }
 
-- (void)refreshContacts {
-    [[ContactsUpdater sharedUpdater] updateSignalContactIntersectionWithABContacts:self.contactsManager.allContacts
-                                                                           success:^{
-                                                                               [self updateContacts];
-                                                                           }
-                                                                           failure:^(NSError *error) {
-                                                                               dispatch_async(dispatch_get_main_queue(), ^{
-                                                                                   UIAlertView *alert =
-                                                                                   [[UIAlertView alloc] initWithTitle:NSLocalizedString(@"ERROR_WAS_DETECTED_TITLE", @"")
-                                                                                                              message:NSLocalizedString(@"TIMEOUT_CONTACTS_DETAIL", @"")
-                                                                                                             delegate:nil
-                                                                                                    cancelButtonTitle:NSLocalizedString(@"OK", @"")
-                                                                                                    otherButtonTitles:nil];
-                                                                                   [alert show];
-                                                                                   [self updateAfterRefreshTry];
-                                                                               });
-                                                                           }];
-    
-    if ([self.contacts count] == 0) {
-        [self showLoadingBackgroundView:YES];
+- (BOOL)isCurrentUserContact:(Contact *)contact
+{
+    for (PhoneNumber *phoneNumber in contact.parsedPhoneNumbers) {
+        if ([[phoneNumber toE164] isEqualToString:[TSAccountManager localNumber]]) {
+            return YES;
+        }
     }
+
+    return NO;
 }
 
-- (void)updateContacts {
-    self.contacts = self.contactsManager.signalContacts;
-    dispatch_async(dispatch_get_main_queue(), ^{
-        [self updateSearchResultsForSearchController:self.searchController];
-        [self.tableView reloadData];
-        [self updateAfterRefreshTry];
-    });
+- (NSArray<Contact *> *_Nonnull)filteredContacts
+{
+    NSMutableArray<Contact *> *result = [NSMutableArray new];
+    for (Contact *contact in self.contactsManager.signalContacts) {
+        if (![self isContactBlocked:contact] && ![self isCurrentUserContact:contact]) {
```

Comment:
Yup, this work is outdated because these PRs have been so backed up.  I think I did most of this work over the weekend.  

There's a trello card to change some of this behavior and a) always show blocked users and self and b) label them as such c) and in some cases force the user to unblock to use them or prevent them from re-blocking a blocked user, etc.  

My original intent was to merge this before making those changes since those changes are not ship blockers and this change is better than nothing IMO.  But I guess I'll just go ahead and make those changes here.